src/protocol/error.js

```diff
@@ -465,13 +465,21 @@ const unknownErrorCode = errorCode => ({
 })
 
 const SUCCESS_CODE = 0
+const UNSUPPORTED_VERSION_CODE = 35
+
 const failure = code => code !== SUCCESS_CODE
 const createErrorFromCode = code => {
   return new KafkaJSProtocolError(errorCodes.find(e => e.code === code) || unknownErrorCode(code))
 }
+const failIfVersionNotSupported = code => {
+  if (code === UNSUPPORTED_VERSION_CODE) {
+    throw new KafkaJSProtocolError(errorCodes.find(e => e.code === code))
```

Comment:
Just delegate this to `createErrorFromCode` instead, so you don't need to replicate the logic for finding the error.